fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/FedoraEventImpl.java

```diff
@@ -227,46 +226,68 @@ public String toString() {
      */
     public static EventType valueOf(final Integer i) {
         final EventType type = translation.get(i);
-        if (isNull(type)) {
+        if (type == null) {
             throw new IllegalArgumentException("Invalid event type: " + i);
         }
         return type;
     }
 
-
     /**
      * Convert a JCR Event to a FedoraEvent
+     *
      * @param event the JCR Event
      * @return a FedoraEvent
      */
     public static FedoraEvent from(final Event event) {
         requireNonNull(event);
         try {
-            @SuppressWarnings("unchecked")
             final Map<String, String> info = event.getInfo();
-
-            return new FedoraEventImpl(valueOf(event.getType()), cleanPath(event),
+            final FedoraEventImpl fedoraEvent = new FedoraEventImpl(valueOf(event.getType()), cleanPath(event),
                     event.getUserID(), event.getUserData(), event.getDate(), info);
-
+            getResourceTypes(event).map(ResourceFactory::createResource).forEach(fedoraEvent::addResourceType);
+            return fedoraEvent;
         } catch (final RepositoryException ex) {
             throw new RepositoryRuntimeException("Error converting JCR Event to FedoraEvent", ex);
         }
     }
 
     /**
-     * The JCR-based Event::getPath contains some Modeshape artifacts that must be removed or modified in
-     * order to correspond to the public resource path. For example, JCR Events will contain a trailing
-     * /jcr:content for Binaries, a trailing /propName for properties, and /#/ notation for URI fragments.
+     * @param event
+     * @return the types recorded on the resource associated to this event
+     */
+    public static Stream<String> getResourceTypes(final Event event) {
+        if (event instanceof org.modeshape.jcr.api.observation.Event) {
+            try {
+                final org.modeshape.jcr.api.observation.Event modeEvent =
+                        (org.modeshape.jcr.api.observation.Event) event;
+                final Stream.Builder<NodeType> types = Stream.builder();
+                for (final NodeType type : modeEvent.getMixinNodeTypes()) {
```

Comment:
No, I thought about that, and if you _really_ want it, fine, but it seems like the creation of several short-lived objects for very little gain. I'm getting more sensitive to that as evidence mounts that stuff coming out of GC eden is a contributor to lower throughput for a lot of code.

I would rather change our code standards to allow the elision of braces for a one-line `for` like this, which would provide as much concision at a lower cost.